src/main/java/com/faforever/client/fa/ForgedAllianceService.java

```diff
@@ -51,6 +53,18 @@ public Process startGameOffline(String map) throws IOException {
   }
 
   public Process startGameOnline(GameLaunchResponse gameLaunchMessage, int gpgPort, int localReplayPort, boolean rehost) throws IOException {
+    LaunchCommandBuilder launchCommandBuilder = prepareLaunchCommand(gameLaunchMessage, gpgPort, localReplayPort, rehost);
+
+    return launch(launchCommandBuilder.build());
+  }
+
+  public Process startGameOnlineWithDivision(GameLaunchResponse gameLaunchMessage, int gpgPort, int localReplayPort, boolean rehost, String divisionName) throws IOException {
+    LaunchCommandBuilder launchCommandBuilder = prepareLaunchCommand(gameLaunchMessage, gpgPort, localReplayPort, rehost);
+
+    return launch(launchCommandBuilder.division(divisionName).build());
+  }
```

Comment:
I wouldn't make a new function and name. Ideally the info would be contained in the gameLaunchMessage. Maybe it might be better to have a new POJO that holds all the information necessary for launching the game that is populated in the game service, rather than using the DTO object.

Reply:
Maybe we can use the existing LaunchCommandBuilder for that? We can set information there in multiple steps if we pass the Builder and only call build() in the ForgedAllianceService when everything is ready. Then we don't need an entirely new class.

Reply:
No I wouldn't use the builder for that. It's purpose is for creating the proper command and setting the variables. Creating a domain object fits the rest of the pattern as well with things like player->PlayerBean